### PR TITLE
Split reports code

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Description

Split Reports code into multiple files to make maintaining it easier.

## Additional details

This is mostly just moving all `class`es defined in `reports.es6` into their separate files. Also cleaned up linter warnings and defined one new constant.

Did it as "nice to have" thing while fixing #3183 (I had to rebuild project multiple times due to Enketo issues so had some time to kill).

### CODE REVIEW

File: jsapp/js/components/reports/reports.es6 [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+  }
+
+  loadReportData() {
+    const uid = this.props.params.assetid || this.props.params.uid;
+
+    stores.allAssets.whenLoaded(uid, (asset) => {
+      let rowsByKuid = {};
+      let rowsByIdentifier = {};
+      let groupBy = '',
+          reportStyles = asset.report_styles,
```
Comment (duvld) [inline jsapp/js/components/reports/reports.es6:64]: it looked like the linting expanded variable declarations from `let x, y` into `let x; let y;`. maybe a missed spot here?
same thing on line 150

File: jsapp/js/components/reports/reportContents.es6 [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        (_type === QUESTION_TYPES.select_one.id || _type === QUESTION_TYPES.select_multiple.id) &&
+        asset.content.choices
+      ) {
+        let question = asset.content.survey.find((z) => {return z.name === _qn || z.$autoname === _qn;});
+        let resps = reportData[i].data.responses;
+        let choice;
+        if (resps) {
+          reportData[i].data.responseLabels = [];
+          for (var j = resps.length - 1; j >= 0; j--) {
+            choice = asset.content.choices.find((o) => {
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/reports/reportContents.es6:76]: hah, still isn't best :D

File: jsapp/js/components/reports/chartColorsPicker.es6 [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ */
+export default class ChartColorsPicker extends React.Component {
+  constructor(props) {
+    super(props);
+    autoBind(this);
+  }
+
+  defaultReportColorsChange(evt) {
+    this.props.onChange(
+      { default: true },
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/reports/chartColorsPicker.es6:18]: Right, so I think that my prettier (I use default with few rules being based on our `.editorconfig`) and our `.eslintrc.json` differ as to spaces in this situation. I was used to removing (as this matches how we handle `(` and `[`, etc.). 

I also have lines mergin in my editor, and when used it always adds a space between merged lines, resulting in `{ default: true },`. I think I merged those and missed that.

Maybe it's a good moment to add `.prettierrc` file to our repo? Not to automatically use it right now, but to decide on what's in it for future.

[the remark this answers is not in the document]
Comment (duvld) [reply jsapp/js/components/reports/chartColorsPicker.es6:18]: > Maybe it's a good moment to add `.prettierrc` file to our repo? Not to automatically use it right now, but to decide on what's in it for future.

Yeah I agree, just so we're consistent with what our linters are doing for new code :D

**Pull request merged**